### ISSUE TITLE
in-addr.arpa zone SOA not working..

### ISSUE THREAD

**Issue by bcookatpcsd** [TechnitiumSoftware/DnsServer#1090]
Am I asking for the SOA type incorrectly.. (don't think so..) 

Agree that the RP came over as invalid.. and I cannot seem to fix.. 

```
 drill -p 530 4.120.10.in-addr.arpa @10.20.112.200 txt -Q
"and .. "
```

```
 drill -p 530 4.120.10.in-addr.arpa @10.20.112.200 soa
;; ->>HEADER<<- opcode: QUERY, rcode: NOERROR, id: 12688
;; flags: qr rd ; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0
;; QUESTION SECTION:
;; 4.120.10.in-addr.arpa.       IN      SOA

;; ANSWER SECTION:

;; AUTHORITY SECTION:

;; ADDITIONAL SECTION:

;; Query time: 0 msec
;; SERVER: 10.20.112.200
;; WHEN: Mon Oct 28 15:39:36 2024
;; MSG SIZE  rcvd: 39
```

![image](https://github.com/user-attachments/assets/9933ecc6-d0f8-49d6-831f-488cea8960ea)

and system logs.. 

![image](https://github.com/user-attachments/assets/5d123e3c-4743-4e85-bcb7-1acef6900d2d)

Thank you in advance for taking the time to read this..

**Comment by ShreyasZare** [comment]
Thanks for the post. It seems that the zone is a Conditional Forwarder zone. The SOA record in it is just a dummy record to allow zone transfer to work when you need to have a Secondary Forwarder zone, so the SOA is not authoritative and cannot be queries or used in any response. 

Usually, Conditional Forwarder zone will contain a FWD record which will fetch the SOA from the upstream server. It does not seem that you have an FWD record in place.

Since this is a reverse zone for private IP network, it would suggest that you use a Primary zone instead of Conditional Forwarder zone.

**Comment by bcookatpcsd** [comment]
So how it seems to work with the TXT entry and other entries (A)..

Anything not defined locally is yes, sent upstream.

Except in this case, the SOA record.

Is that how it also looks?

"When would I not send it upstream?"

"When I have a local answer."

If none of the local answers are allowed, I don't think it would make sense to allow them to be created.

Except it does appear that it returns things define locally and then send upstream if not found..

Yes?

**Comment by ShreyasZare** [comment]
Yes, it will use records in the zone to answer and for anything that does not exists, it will fetch from the upstream/forwarder defined in the FWD record. 

But since the SOA record, that you see in the forwarder zone, is a dummy record, it does not really exists from the DNS query perspective and its used only for zone transfer.
